### PR TITLE
Fixes magic mock import issues in multiple versions of python

### LINKED ISSUE #1549: Nightly downstream job failing with LinchpinAPI import errors
**Describe the bug**
Our jobs are failing with the failing with the following error

```
 File "/home/jenkins/workspace/Carbon-Linchpin-Nightly/cbn/lib/python2.7/site-packages/carbon/provisioners/ext/linchpin_wrapper_plugin.py", line 83, in __init__
    self.linchpin_api = LinchpinAPI(self._init_context())
NameError: global name 'LinchpinAPI' is not defined
```

This is using latest development branch and python 2.7. 

If open up a python console and try to perform the same import, I get the following error messages about the mock library, which I think is causing the import of the API to not work. 

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/dbaez/.virtualenvs/linchpin-nightly/lib/python2.7/site-packages/linchpin/__init__.py", line 20, in <module>
    from mock import MagicMock
ImportError: No module named mock
```
 
**To Reproduce**
1. Install latest develop branch `pip install git+https://github.com/CentOS-PaaS-SIG/linchpin.git@develop`
2. open python console `python`
3. import the LinchpinAPI `from linchpin import LinchpinAPI`

### PR DESCRIPTION
**Pull request by samvarankashyap**
fixes #1549

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -23,4 +23,5 @@ future
 openstacksdk>=0.37
 tqdm==4.36.1
 pyzmq==18.1.1
+mock
```
Comment (14rcole) [inline requirements.txt]: Can this be placed in the test requirements instead?  It's not really useful for ordinary users

Comment (abraverm) [reply requirements.txt]: It is actually required for disabling progress bar and multiprocessing.

Comment (14rcole) [reply requirements.txt]: I'm sorry, I thought this was related to the mock tests

**Comment by samvarankashyap** [comment]
[merge]

**Pull request merged**